src/main/java/core/basesyntax/strategy/impl/BalanceOperation.java

```diff
@@ -0,0 +1,25 @@
+package core.basesyntax.strategy.impl;
+
+import core.basesyntax.dao.StorageDao;
+import core.basesyntax.model.Fruit;
+import core.basesyntax.model.FruitTransaction;
+import core.basesyntax.strategy.Operation;
+import core.basesyntax.strategy.OperationException;
+
+public class BalanceOperation implements Operation {
+    private final StorageDao storageDao;
+
+    public BalanceOperation(StorageDao storageDao) {
+        this.storageDao = storageDao;
+    }
+
+    @Override
+    public void proceed(FruitTransaction fruitTransaction) {
+        if (storageDao.get(fruitTransaction.getFruitName()) != null) {
+            throw new OperationException("Fruit " + fruitTransaction.getFruitName()
+                    + " already exist");
```

Comment:
According to the checklist, the balance operation should set the quantity of fruit in the storage. Throwing an exception if the fruit already exists is not the correct behavior for a balance operation. The balance operation should update the storage with the new quantity provided, regardless of whether the fruit already exists or not. You should replace the exception with logic to set the quantity of the fruit in the storage.